pkg/actuators/machine/utils.go

```diff
@@ -398,3 +403,97 @@ func GetMibValueOfQuantity(quantity resource.Quantity) int64 {
 func Ptr[T any](v T) *T {
 	return &v
 }
+
+// GetGPU returns the VMGpu for the given Prism Element (uuid) and GPU identifier input
+func GetGPU(ctx context.Context, client *nutanixClientV3.Client, peUUID string, gpu machinev1.NutanixGPU) (*nutanixClientV3.VMGpu, error) {
+	peGPUs, err := GetGPUsForPE(ctx, client, peUUID)
+	if err != nil {
+		return nil, err
+	}
+	if len(peGPUs) == 0 {
+		return nil, fmt.Errorf("no available GPUs found in Prism Element cluster with UUID %s", peUUID)
+	}
+
+	foundGPU, err := GetGPUFromList(ctx, client, gpu, peGPUs)
+	if err != nil {
+		return nil, err
+	}
+
+	return foundGPU, nil
+}
+
+// GetGPUList returns a list of VMGpus for the given list of GPU identifiers in the Prism Element (uuid)
+func GetGPUList(ctx context.Context, client *nutanixClientV3.Client, gpus []machinev1.NutanixGPU, peUUID string) ([]*nutanixClientV3.VMGpu, error) {
```

Comment:
As above, does this really need to be public?

Reply:
Changed the method name to start with lower case.